### ISSUE TITLE
Version 0.6.2 doesn't parse nested wikilinks

### ISSUE THREAD

**Issue by gamecat10** [earwig/mwparserfromhell#270]
v. 0.6
```
import mwparserfromhell as mw
a='[[File:test|[[test]]]]'
type(mw.parse(a).nodes[0])
<class 'mwparserfromhell.nodes.wikilink.Wikilink'>

```

v. 0.6.2:
```
import mwparserfromhell as mw
a='[[File:test|[[test]]]]'
type(mw.parse(a).nodes[0])
<class 'mwparserfromhell.nodes.text.Text'>
```

**Comment by earwig** [comment]
Thanks. This is indeed a regression in 0.6.2. I'll work out a fix for this.

**Comment by demostanis** [comment]
bump pls im having a headache

**Comment by geohci** [comment]
I wanted to check in on this too -- is there a sense of where the issue is / how fixable it is? I assume it's on the C side and not Python (which limits my ability to help greatly) but am happy to try to help if need be.

**Comment by earwig** [comment]
Thanks for offering to help! The parser has a Python reference implementation; I'm happy to accept PRs for fixes only made there which I can then port to the C parser.

However, I'll have some time later this week to investigate, so hopefully that's not necessary.

**Comment by geohci** [comment]
Many thanks! I'll wait for your initial investigation then.

Unrelated but huge thanks for this work -- I've used mwparserfrom for a while for all sorts of tasks. At the moment, I'm working on a Python package for describing what happens in a given Wikipedia diff (e.g., 3 templates changed, 2 categories added, etc.) and mwparserfromhell has been so powerful and useful -- this is a bug I've stumbled across but it's very minor in the grand scheme of things.
